src/write.c

```diff
@@ -726,6 +727,102 @@ static avifResult avifEncoderDataCreateXMPItem(avifEncoderData * data, const avi
     return AVIF_RESULT_OK;
 }
 
+// Copies the pixel from srcPlane to the already allocated dstPlane,
+// filling any extra row or column with border pixel values.
+static void avifPlaneCopyAndPad(uint8_t * dstPlane,
+                                uint32_t dstRowBytes,
+                                uint32_t dstWidth,
+                                uint32_t dstHeight,
+                                const uint8_t * srcPlane,
+                                uint32_t srcRowBytes,
+                                uint32_t srcWidth,
+                                uint32_t srcHeight,
+                                uint32_t sampleByteCount)
+{
+    for (uint32_t j = 0; j < srcHeight; ++j) {
```

Comment:
are there checks anywhere to ensure dstWidth/Height are >= src? If so, then maybe an assert can be added here

Reply:
> are there checks anywhere to ensure dstWidth/Height are >= src?

Yes, at line 885:

```c
    if ((topRightCell->width > tileWidth) || (bottomLeftCell->height > tileHeight)) {
        return AVIF_RESULT_INVALID_IMAGE_GRID;
    }
```

> If so, then maybe an assert can be added here

Done.